### PR TITLE
Procedure and documentation for release process

### LINKED ISSUE #330: Establish release process procedure
We should establish reliable release process to be included as a default addon for K8s clusters. Here's my proposal. 

For each release:

1. Send a pull request that increases version number in `build/conf.js`. Follow semver versioning: http://semver.org/. 
1. The pull request should contain release notes written in `CHANGELOG.md`. 
1. Get it reviewed and merged.
1. Create a git tag for the merged PR. With the name the same as the version.
1. Build and push images to GCR with `gulp push-to-gcr`.
1. Update addons configuration in k8s core.

### PR DESCRIPTION
**Pull request by bryk**
Also, minor cleanup of other docs.

Fixes #330

### CODE REVIEW

**Comment by bryk** [comment]
@cheld @floreks Can you take a look?

File: docs/devel/release.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+1. Send a pull request that increases Kubernetes Dashboard version number in `build/conf.js`.
+   The property name is `deploy.versionRelease`. Follow versioning guidelines.
+1. Get the pull request reviewed and merged.
+1. Create a git [release tag](https://github.com/kubernetes/dashboard/releases/) for the merged
+   pull request. Release description should include changelog.
+1. Build and push production images to container registry. Use `gulp push-to-gcr:release`.
+1. Update addons on the Kubernetes core repository. Dashboard addon directory is
+   [here](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/dashboard). If
+   the update is minor, all that needs to be done is to change image version number in the main
+   controller config file (and other configs, as described in the header of the config).
```
Comment (cheld) [inline docs/devel/release.md]: main controller config file - I could not find this file...

LGTM

Comment (bryk) [inline docs/devel/release.md]: Yeah, fixed. I've named it explicitly now.

**Comment by bryk** [comment]
Thanks for review. Will merge soon.

**Comment by cheld** [comment]
It seems to me that you have only updated GCE provider. Have added Dashboard to vagrant provider?

**Comment by bryk** [comment]
I haven't added it to the Vagrant provider. Can you send me a link to the config files?

**Comment by cheld** [comment]
I just wanted to understand what you have done so far. If I remember correctly the switches for add-ons are in this file:

https://github.com/cheld/kubernetes/blob/master/cluster/vagrant/config-default.sh

However, I think you can leave the flag "ENABLE_CLUSTER_UI" as it is. It is from the previous kube-ui.

The deployment of the pod should be done by salt. 
https://github.com/cheld/kubernetes/blob/master/cluster/saltbase/salt/kube-addons/init.sls

My guess would be that you have only to change the upper file from kube-ui paths to dashboard-path. 
This would affect all other providers that use the salt-scripts

**Comment by bryk** [comment]
Can you do the change for vagrant (or tell somebody from your team) and CC me? I want to make sure others get external visibility too.

**Comment by cheld** [comment]
Ok, I give it a try. If I am too busy with non-serious work than I will pass it on

**Pull request merged**